exp/api/v1beta2/aromachinepool_types.go

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+)
+
+// AROMachinePoolSpec defines the desired state of AROMachinePool.
+type AROMachinePoolSpec struct {
+	// NodePoolName specifies the name of the nodepool in ARO.
+	// It must be a valid DNS-1035 label, so it must consist of lower case alphanumeric characters and have a maximum length of 15 characters.
+	//
+	// +immutable
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="nodepoolName is immutable"
+	// +kubebuilder:validation:MaxLength:=15
+	// +kubebuilder:validation:Pattern:=`^[a-z]([-a-z0-9]*[a-z0-9])?$`
+	NodePoolName string `json:"nodePoolName"`
+
+	// Version specifies the OpenShift version of the nodes associated with this machinepool.
+	// AROMachinePool version is used if not set.
+	//
+	// +optional
+	Version string `json:"version,omitempty"`
+
+	// OpenShift version channel group; default is stable.
+	//
+	// +kubebuilder:validation:Enum=stable;candidate;nightly
```

Comment:
no channel group for machinePool, it should follow the AroControlPlane channelGroup